### PR TITLE
Fix h5netcdf deprecation warning

### PR DESCRIPTION
**Pull request by dstansby**
As far as I can tell from the h5necdf documentation we want to decode strings here. This should fix the CI.

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Looks like there is another failure: 

```python
_____________________________________________________________________ TestTimeSeries.test_implicit_new_goes15 ______________________________________________________________________
[gw2] linux -- Python 3.9.1 /home/vsts/work/1/s/.tox/py39/bin/python

self = <sunpy.timeseries.tests.test_timeseries_factory.TestTimeSeries object at 0x7f5ceba92970>

    def test_implicit_new_goes15(self):
        # Test a GOES TimeSeries
>       ts_goes = sunpy.timeseries.TimeSeries(new_goes15_filepath)

../../.tox/py39/lib/python3.9/site-packages/sunpy/timeseries/tests/test_timeseries_factory.py:128: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py39/lib/python3.9/site-packages/sunpy/timeseries/timeseries_factory.py:406: in __call__
    new_ts = self._check_registered_widgets(filepath=filepath, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sunpy.timeseries.timeseries_factory.TimeSeriesFactory object at 0x7f5ced9d09d0>, kwargs = {}, WidgetType = <class 'sunpy.timeseries.timeseriesbase.GenericTimeSeries'>
filepath = '/home/vsts/work/1/s/.tox/py39/lib/python3.9/site-packages/sunpy/data/test/goes_truncated_test_goes15.nc', data = None, meta = None, units = None

    def _check_registered_widgets(self, **kwargs):
        """
        Checks the (instrument) source(s) that are compatible with this given
        file/data.
    
        Only if exactly one source is compatible will a
        `~sunpy.timeseries.TimeSeries` be returned.
        """
        WidgetType = self._get_matching_widget(**kwargs)
    
        # Dealing with the fact that timeseries filetypes are less consistent
        # (then maps), we use a _parse_file() method embedded into each
        # instrument subclass.
        filepath = kwargs.pop('filepath', None)
        data = kwargs.pop('data', None)
        meta = kwargs.pop('meta', None)
        units = kwargs.pop('units', None)
        if filepath:
>           data, meta, units = WidgetType._parse_file(filepath)
E           TypeError: cannot unpack non-iterable NotImplementedType object

../../.tox/py39/lib/python3.9/site-packages/sunpy/timeseries/timeseries_factory.py:529: TypeError
----------------------------------------------------- generated xml file: /home/vsts/work/1/s/.tmp/py39/junit/test-results.xml -----------------------------------------------------

```

File: sunpy/timeseries/sources/goes.py [hunk 1]
```diff
@@ -56,6 +57,10 @@ class XRSTimeSeries(GenericTimeSeries):
     # Class attribute used to specify the source class of the TimeSeries.
     _source = 'xrs'
 
+    _netcdf_read_kw = {}
+    if h5netcdf.__version__ >= LooseVersion("0.9"):
+        _netcdf_read_kw['decode_strings'] = True
```
Comment (dstansby) [inline sunpy/timeseries/sources/goes.py:62]: Not sure if this is the best place to put these lines, other suggestions welcome.

Comment (Cadair) [reply sunpy/timeseries/sources/goes.py:63]: Why not just no this in the `_parse_netcdf` method?

Comment (dstansby) [reply sunpy/timeseries/sources/goes.py:63]: That's what I originally did, but it's needed in `is_datasource_for` too.

**Comment by nabobalis** [comment]
Helioviewer is having issues, we can ignore it.

**Pull request merged**